Sources/StreamVideo/WebRTC/WebRTCClient.swift

```diff
@@ -198,15 +213,46 @@ class WebRTCClient: NSObject {
         log.debug("Connecting to SFU", subsystems: .webRTC)
         await state.update(connectionState: .connecting)
         log.debug("Setting user media", subsystems: .webRTC)
-        await setupUserMedia(callSettings: callSettings)
-        log.debug("Connecting WS channel", subsystems: .webRTC)
-        signalChannel?.connect()
-        sfuMiddleware.onSocketConnected = handleOnSocketConnected
+        if !migrating {
+            await setupUserMedia(callSettings: callSettings)
+            log.debug("Connecting WS channel", subsystems: .webRTC)
+            signalChannel?.connect()
+            sfuMiddleware.onSocketConnected = handleOnSocketConnected
+        } else {
+            log.debug("Performing session migration", subsystems: .webRTC)
+            migratingWSClient?.connect()
```

Comment:
shouldn't we do a check here if `migratingWSClient` and `publisher` are not `nil` in order to avoid confusion on our logs? (if the `migratingWSClient` is nil we cannot perform a session migration, right?)

Reply:
`publisher` can actually be `nil`. This happens when you are not publishing video nor audio during a call, so that one is not mandatory.
As for `migratingWSClient`, that one is always setup when `migratingFrom` is not nil. If it fails, we will still use the old one if possible, but it's good to have a log that we tried to migrate.